src/components/PostDetails.tsx

```diff
@@ -2,105 +1,120 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment } from '../api/comments';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  selectedPost: Post | null;
+  comments: Comment[];
+  setComments: (comment: Comment[]) => void;
+  loading: boolean;
+  isError: string;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  comments,
+  setComments,
+  loading,
+  isError,
+}) => {
+  const [form, setForm] = useState(false);
+
+  useEffect(() => {
+    setForm(false);
+  }, [selectedPost]);
+
+  function deleteCurrComment(commentId: number) {
+    const updatedComments = comments.filter(com => com.id !== commentId);
+
+    setComments(updatedComments);
+
+    return deleteComment(commentId).catch(error => {
+      setComments(comments);
+      throw error;
```

Comment:
Consider providing user feedback when an error occurs during comment deletion. This can improve user experience by informing users of the failure and potentially allowing them to retry.